spring-restdocs-core/src/main/java/org/springframework/restdocs/payload/RequestPartFieldsSnippet.java

```diff
@@ -0,0 +1,180 @@
+	 * failure.
+	 *
+	 * @param partName    the part name
+	 * @param descriptors the descriptors
+	 */
+	protected RequestPartFieldsSnippet(String partName, List<FieldDescriptor> descriptors) {
+		this(partName, descriptors, null, false);
+	}
+
+	/**
+	 * Creates a new {@code RequestPartFieldsSnippet} that will document the fields in the
+	 * request part using the given {@code descriptors}. If {@code ignoreUndocumentedFields} is
+	 * {@code true}, undocumented fields will be ignored and will not trigger a failure.
+	 *
+	 * @param partName                 the part name
+	 * @param descriptors              the descriptors
+	 * @param ignoreUndocumentedFields whether undocumented fields should be ignored
+	 */
+	protected RequestPartFieldsSnippet(String partName, List<FieldDescriptor> descriptors,
+									boolean ignoreUndocumentedFields) {
+		this(partName, descriptors, null, ignoreUndocumentedFields);
+	}
+
+	/**
+	 * Creates a new {@code RequestFieldsSnippet} that will document the fields in the
+	 * request using the given {@code descriptors}. The given {@code attributes} will be
+	 * included in the model during template rendering. Undocumented fields will trigger a
+	 * failure.
+	 *
+	 * @param partName    the part name
+	 * @param descriptors the descriptors
+	 * @param attributes  the additional attributes
+	 */
+	protected RequestPartFieldsSnippet(String partName, List<FieldDescriptor> descriptors,
+									Map<String, Object> attributes) {
+		this(partName, descriptors, attributes, false);
+	}
+
+	/**
+	 * Creates a new {@code RequestFieldsSnippet} that will document the fields in the
+	 * request using the given {@code descriptors}. The given {@code attributes} will be
+	 * included in the model during template rendering. If
+	 * {@code ignoreUndocumentedFields} is {@code true}, undocumented fields will be
+	 * ignored and will not trigger a failure.
+	 *
+	 * @param partName                 the part name
+	 * @param descriptors              the descriptors
+	 * @param attributes               the additional attributes
+	 * @param ignoreUndocumentedFields whether undocumented fields should be ignored
+	 */
+	protected RequestPartFieldsSnippet(String partName, List<FieldDescriptor> descriptors,
+									Map<String, Object> attributes, boolean ignoreUndocumentedFields) {
+		super("request", descriptors, attributes, ignoreUndocumentedFields);
+		this.partName = partName;
+	}
+
+	@Override
+	protected MediaType getContentType(Operation operation) {
+		for (OperationRequestPart candidate : operation.getRequest().getParts()) {
+			if (candidate.getName().equals(this.partName)) {
+				candidate.getHeaders().getContentType();
+			}
+		}
+		return null;
+	}
+
+	@Override
+	protected byte[] getContent(Operation operation) throws IOException {
+		for (OperationRequestPart candidate : operation.getRequest().getParts()) {
+			if (candidate.getName().equals(this.partName)) {
+				return candidate.getContent();
+			}
+		}
+		return new byte[0];
```

Comment:
I wonder if this should fail fast here rather than failing later when a documented field in the part isn't found?